geopandas/tools/geocoding.py

```diff
@@ -22,7 +23,12 @@ def _get_throttle_time(provider):
         return 0
 
 
-def geocode(strings, provider=None, **kwargs):
+def geocode(
+    strings: list[str] | pd.Series,
+    provider: str | "geopy.geocoder" = None,
+    throttle_time: int | float = None,
+    **kwargs,
```

Comment:
`RateLimiter` has some parameters.
I thought `max_retries` and `error_wait_seconds` optional arguments could expose to `geocoder.(geocode|reverse_geocode)`.

Reply:
I would maybe add `limiter_kwargs` keyword to expose `RateLimiter` kwargs.

Reply:
If use `limiter_kwargs`, there is a problem with `min_delay_seconds`.
`min_delay_seconds` is coming from `RateLimiter`.
Wether should `min_delay_seconds` be directly exposed to `geocode` and `reverse_geocode`?
We could use `limiter_kwargs` via `limiter_kwargs=dict(min_delay_seconds=...)`.

[RateLimiter's parameters](https://geopy.readthedocs.io/en/stable/#geopy.extra.rate_limiter.RateLimiter.__init__)

```python
class geopy.extra.rate_limiter.RateLimiter(func, *, min_delay_seconds=0.0, max_retries=2, error_wait_seconds=5.0, swallow_exceptions=True, return_value_on_exception=None)
```